aws-lc-rs/tests/rsa_test.rs

```diff
@@ -340,3 +425,406 @@ fn public_key_components_clone_debug() {
+
+                let ciphertext = public_key
+                    .encrypt($alg, MESSAGE, ciphertext.as_mut(), Some(&[0u8; 0]))
+                    .expect("encrypted");
+
+                let mut plaintext = vec![0u8; private_key.min_output_size()];
+
+                let plaintext = private_key
+                    .decrypt($alg, ciphertext, &mut plaintext, Some(&[0u8; 0]))
+                    .expect("decrypted");
+
+                assert_eq!(MESSAGE, plaintext);
+            }
+
+            // fixed message, Some(label)
+            {
+                let mut ciphertext = vec![0u8; public_key.ciphertext_size()];
+
+                let label: &[u8] = br"Testing Data Label";
+
+                let ciphertext = public_key
+                    .encrypt($alg, MESSAGE, ciphertext.as_mut(), Some(label))
+                    .expect("encrypted");
+
+                let mut plaintext = vec![0u8; private_key.min_output_size()];
+
+                let plaintext = private_key
+                    .decrypt($alg, ciphertext, &mut plaintext, Some(label))
+                    .expect("decrypted");
+
+                assert_eq!(MESSAGE, plaintext);
+            }
+
+            // zero-length message
+            {
+                let message: &[u8] = &[1u8; 0];
+                let mut ciphertext = vec![0u8; public_key.ciphertext_size()];
+
+                let ciphertext = public_key
+                    .encrypt($alg, message, ciphertext.as_mut(), None)
+                    .expect("encrypted");
+
+                let mut plaintext = vec![0u8; private_key.min_output_size()];
+
+                let plaintext = private_key
+                    .decrypt($alg, ciphertext, &mut plaintext, None)
+                    .expect("decrypted");
+
+                assert_eq!(message, plaintext);
+            }
+
+            // max_plaintext_size message
+            {
+                let message = vec![1u8; public_key.max_plaintext_size($alg)];
+                let mut ciphertext = vec![0u8; public_key.ciphertext_size()];
+
+                let ciphertext = public_key
+                    .encrypt($alg, &message, ciphertext.as_mut(), None)
+                    .expect("encrypted");
+
+                let mut plaintext = vec![0u8; private_key.min_output_size()];
+
+                let plaintext = private_key
+                    .decrypt($alg, ciphertext, &mut plaintext, None)
+                    .expect("decrypted");
+
+                assert_eq!(message, plaintext);
+            }
+
+            // max_plaintext_size+1 message
+            {
+                let message = vec![1u8; public_key.max_plaintext_size($alg) + 1];
+                let mut ciphertext = vec![0u8; private_key.min_output_size()];
+
+                public_key
+                    .encrypt($alg, &message, ciphertext.as_mut(), None)
+                    .expect_err("plaintext too large");
+            }
+        }
+    };
+}
+
+round_trip_algorithm!(
```

Comment:
NP: nested loops? Or are there some invalid combinations?

Reply:
The macro is preferred as it results in a test function per combination, versus a loop would bail out the test earlier and you don't get a full picture of the failures.